.github/workflows/release.yml

```diff
@@ -0,0 +1,341 @@
+          echo "All individual ZIP archives created:"
+          ls -lh *.zip
+
+      - name: Generate SHA256SUMS
+        run: |
+          cd bin
+
+          # Generate checksums for all files including individual ZIPs and all-platforms ZIP
+          sha256sum terragrunt_* > SHA256SUMS
+
+          echo "SHA256SUMS generated:"
+          cat SHA256SUMS
+
+          echo ""
+          echo "Total files with checksums: $(wc -l < SHA256SUMS)"
+
+      - name: Verify signatures before upload
+        run: |
+          echo "Verifying required files..."
+
+          # Check macOS binaries
+          for file in terragrunt_darwin_amd64 terragrunt_darwin_arm64; do
+            if [ -f "bin/$file" ]; then
+              echo "$file present"
+            else
+              echo "$file missing"
+              exit 1
+            fi
+          done
+
+          # Check Windows binaries
+          for file in terragrunt_windows_amd64.exe terragrunt_windows_386.exe; do
+            if [ -f "bin/$file" ]; then
+              echo "$file present"
+            else
+              echo "$file missing"
+              exit 1
+            fi
+          done
+
+          # Check Linux binaries
+          for file in terragrunt_linux_386 terragrunt_linux_amd64 terragrunt_linux_arm64; do
+            if [ -f "bin/$file" ]; then
+              echo "$file present"
+            else
+              echo "$file missing"
+              exit 1
+            fi
+          done
+
+          # Check SHA256SUMS
+          if [ -f "bin/SHA256SUMS" ]; then
+            echo "SHA256SUMS present"
+          else
+            echo "SHA256SUMS missing"
+            exit 1
+          fi
+
+          echo "All required files verified"
+
+      - name: Upload assets to release
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          VERSION: ${{ steps.version.outputs.version }}
+        run: |
+          echo "Uploading assets to existing release $VERSION..."
+
+          # Upload all files using gh CLI
+          cd bin
+          for file in *; do
+            echo "Uploading $file..."
+            if gh release upload "$VERSION" "$file" --clobber; then
+              echo "Uploaded $file"
+            else
+              echo "Upload failed for $file (will retry in verification)"
+            fi
+          done
+
+          echo "Upload phase completed"
+
+      - name: Verify all assets uploaded
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          VERSION: ${{ steps.version.outputs.version }}
+        run: |
+          MAX_RETRIES=10
+
+          echo "Verifying all assets are accessible..."
+
+          # Get list of assets in the release
+          ASSETS=$(gh release view "$VERSION" --json assets --jq '.assets[].name')
+          asset_count=$(echo "$ASSETS" | wc -l)
+
+          echo "Found $asset_count assets in release"
+
+          # Expected files (binaries + individual ZIPs + SHA256SUMS)
+          expected_files=(
```

Comment:
Can we derive this from a single source of truth? I feel like we have to synchronize a lot of different filename lists right now.